### PR TITLE
Bump stage0 compiler to 1.56

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
r? @pietroalbini (but others should feel free to steal)

### CODE REVIEW

File: library/std/src/primitive_docs.rs [hunk 1]
```diff
@@ -612,8 +612,7 @@ mod prim_pointer {}
 /// Starting in the 2021 edition, `array.into_iter()` uses `IntoIterator` normally to iterate
 /// by value, and `iter()` should be used to iterate by reference like previous editions.
 ///
-#[cfg_attr(bootstrap, doc = "```rust,edition2021,ignore")]
-#[cfg_attr(not(bootstrap), doc = "```rust,edition2021")]
+/// ```rust,edition2021,ignore
```
Comment (m-ou-se) [inline library/std/src/primitive_docs.rs:615]:
```suggestion
/// ```rust,edition2021
```

**Comment by m-ou-se** [comment]
Yess, time to bump the compiler and library to `edition = "2021"`!

@bors r+

**Comment by pietroalbini** [comment]
Also looks good to me!

**Comment by Mark-Simulacrum** [comment]
The regressions are pretty small, and scattered in +/- about equally. This PR also at least in theory doesn't change the compiler or standard library being benchmarked, so the differences are likely due to "noise" (presumably related to changes to the memory layout or something similar).

@rustbot label: +perf-regression-triaged

**Pull request merged**